File: app/src/test/java/org/metabrainz/mobile/EntityModelsTest.java
package org.metabrainz.mobile;

import com.google.gson.Gson;

import org.junit.Test;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.Artist;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.Label;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.Recording;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.Release;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.ReleaseGroup;

import static org.metabrainz.mobile.AssertionUtils.checkArtistAssertions;
import static org.metabrainz.mobile.AssertionUtils.checkLabelAssertions;
import static org.metabrainz.mobile.AssertionUtils.checkRecordingAssertions;
import static org.metabrainz.mobile.AssertionUtils.checkReleaseAssertions;
import static org.metabrainz.mobile.AssertionUtils.checkReleaseGroupAssertions;
import static org.metabrainz.mobile.EntityTestUtils.getTestArtist;
import static org.metabrainz.mobile.EntityTestUtils.getTestLabel;
import static org.metabrainz.mobile.EntityTestUtils.getTestRecording;
import static org.metabrainz.mobile.EntityTestUtils.getTestRelease;
import static org.metabrainz.mobile.EntityTestUtils.getTestReleaseGroup;
import static org.metabrainz.mobile.EntityTestUtils.loadResourceAsString;

public class EntityModelsTest {

    @Test
    public void testArtistModel() {
        Artist testArtist = getTestArtist();
        Artist artist = new Gson().fromJson(loadResourceAsString("artist_lookup.json"), Artist.class);
        checkArtistAssertions(testArtist, artist);
    }

    @Test
    public void testReleaseModel() {
        Release testRelease = getTestRelease();
        Release release = new Gson().fromJson(loadResourceAsString("release_lookup.json"), Release.class);
        checkReleaseAssertions(testRelease, release);
    }

    @Test
    public void testReleaseGroupModel() {
        ReleaseGroup testReleaseGroup = getTestReleaseGroup();
        ReleaseGroup releaseGroup = new Gson().fromJson(loadResourceAsString("release-group_lookup.json"), ReleaseGroup.class);
        checkReleaseGroupAssertions(testReleaseGroup, releaseGroup);
    }

    @Test
    public void testLabelModel() {
        Label testLabel = getTestLabel();
        Label label = new Gson().fromJson(loadResourceAsString("label_lookup.json"), Label.class);
        checkLabelAssertions(testLabel, label);
    }

    @Test
    public void testRecordingModel() {
        Recording testRecording = getTestRecording();
        Recording recording = new Gson().fromJson(loadResourceAsString("recording_lookup.json"), Recording.class);
        checkRecordingAssertions(testRecording, recording);
    }

}

File: app/src/test/java/org/metabrainz/mobile/EntityModelsTest.kt
package org.metabrainz.mobile

import com.google.gson.Gson
import org.junit.Test
import org.metabrainz.mobile.EntityTestUtils.loadResourceAsString
import org.metabrainz.mobile.EntityTestUtils.testArtist
import org.metabrainz.mobile.EntityTestUtils.testLabel
import org.metabrainz.mobile.EntityTestUtils.testRecording
import org.metabrainz.mobile.EntityTestUtils.testRelease
import org.metabrainz.mobile.EntityTestUtils.testReleaseGroup
import org.metabrainz.mobile.data.sources.api.entities.mbentity.*

class EntityModelsTest {
    @Test
    fun testArtistModel() {
        val artist: Artist = Gson().fromJson(loadResourceAsString("artist_lookup.json"),Artist::class.java)
        AssertionUtils.checkArtistAssertions(testArtist, artist)
    }

    @Test
    fun testReleaseModel() {
        val release: Release = Gson().fromJson(
            loadResourceAsString("release_lookup.json"),
            Release::class.java)
        AssertionUtils.checkReleaseAssertions(testRelease, release)
    }

    @Test
    fun testReleaseGroupModel() {
        val releaseGroup: ReleaseGroup = Gson().fromJson(
            loadResourceAsString("release-group_lookup.json"),
            ReleaseGroup::class.java
        )
        AssertionUtils.checkReleaseGroupAssertions(testReleaseGroup, releaseGroup)
    }

    @Test
    fun testLabelModel() {
        val label: Label = Gson().fromJson(
            loadResourceAsString("label_lookup.json"),
            Label::class.java)
        AssertionUtils.checkLabelAssertions(testLabel, label)
    }

    @Test
    fun testRecordingModel() {
        val recording: Recording = Gson().fromJson(
            loadResourceAsString("recording_lookup.json"),
            Recording::class.java)
        AssertionUtils.checkRecordingAssertions(testRecording, recording)
    }
}